### PR TITLE
coro: don't probe, just try to find

### PR DESCRIPTION
**Pull request by paboldin**
Don't probe for coroutine finder, just ask them to find coroutines.
They can return CORO_SEARCH_NEXT for us to try next coroutine finder
or return anything else to indicate for success or failure.

Signed-off-by: Pavel Boldin <pboldin@cloudlinux.com>

### CODE REVIEW

**Comment by rashchupkinr** [comment]
Don't see anything wrong.

File: src/kpatch_coro.c [hunk 1]
```diff
@@ -562,41 +574,34 @@ int kpatch_init_coroutine(struct kpatch_process *proc)
...
+
+	for (i = 0; i < ARRAY_SIZE(kpatch_coro_flavours); i++) {
+		struct kpatch_coro_ops *ops = &kpatch_coro_flavours[i];
+
+		rv = ops->find_coroutines(proc);
+		if (rv == CORO_SEARCH_NEXT)
+			continue;
+
+		return rv;
+	}
```
Comment (pborzenkov) [inline src/kpatch_coro.c:614]: I'm thinking, may we should do here something like this:

if (rc == CORO_SEARCH_NEXT && process_is_known_to_use_coroutines()) {
  log("Warning: process contains coroutines but kcare doesn't know how to handle them")
  if (!force)
    exit()
}

And keep a list of processes we know for sure use coroutines, like qemu.

Comment (paboldin) [reply src/kpatch_coro.c:614]: Updated, please review.

Comment (paboldin) [reply src/kpatch_coro.c:614]: I will switch to checking whether `makecontext`, `setjmp` and so on are used by the process.

**Review by pborzenkov — APPROVED** [review]

**Pull request merged**